src/Transport/HttpTransport.php

```diff
@@ -178,17 +178,23 @@ public function delete(string $path): ResponseInterface
     /**
      * @param                                 $path
      * @param string|resource|StreamInterface $file
+     * @param string|null                     $fileName
      * @param array                           $additionalParameters
      *
      * @return ResponseInterface
      */
-    public function upload($path, $file, array $additionalParameters = []): ResponseInterface
+    public function upload($path, $file, string $fileName = null, array $additionalParameters = []): ResponseInterface
     {
         $builder = new MultipartStreamBuilder($this->getStreamFactory());
         foreach ($additionalParameters as $parameter => $value) {
             $builder->addResource($parameter, strval($value));
         }
-        $builder->addResource('file', $file);
+
+        $resourceOptions = [];
+        if($fileName === null) {
```

Comment:
This seems to be wrong

Reply:
It does indeed... Sorry for that, this should be fixed now.